flink-table/flink-sql-parser/src/main/codegen/includes/parserImpls.ftl

```diff
@@ -1711,7 +1711,12 @@ SqlEndStatementSet SqlEndStatementSet() :
 
 /**
 * Parse a statement set.
-* END;
+*
+* STATEMENT SET BEGIN (RichSqlInsert();)+ END
+*
+* Note that when using this parser directive with OrderedQueryOrExpr,
+* you might have choice conflicts because STATEMENT is not a reserved keyword,
```

Comment:
Can we make it a reserved keyword? I don't see a reason why not. We have many reserved keywords and users can simply escape them.